Generating a witness of an enum type produces a compile error in the source generated code

When I add this to the `TypeShape.Tests` project:

```cs
[GenerateShape<MyEnum>]
partial class Witness { }
```

I get this compile error:

>Q:\source\typeshape-csharp\tests\TypeShape.Tests\obj\Debug\net8.0\generated\TypeShape.SourceGenerator\TypeShape.SourceGenerator.TypeShapeIncrementalGenerator\TypeShape.Tests.MsgPackSerializerTests.Witness.MyEnum.g.cs(25,38,25,43): error CS0119: 'int' is a type, which is not valid in the given context

which points to this generated code:

```cs
private global::TypeShape.Abstractions.ITypeShape<global::TypeShape.Tests.MyEnum> Create_MyEnum()
{
    return new global::TypeShape.SourceGenModel.SourceGenEnumTypeShape<global::TypeShape.Tests.MyEnum, int>
    {
        UnderlyingType = Int32, // error on this line
        Provider = this,
    };
}
```

I can workaround it by adding yet another witness explicitly:

```diff
+[GenerateShape<int>]
 [GenerateShape<MyEnum>]
 partial class Witness { }
```

But having to do that seems very tedious. Shouldn't transitive types like this be implicit?

--- a/src/TypeShape.Roslyn/ModelGenerator/TypeDataModelGenerator.Enum.cs
+++ b/src/TypeShape.Roslyn/ModelGenerator/TypeDataModelGenerator.Enum.cs
@@ -1,10 +1,11 @@
-﻿using Microsoft.CodeAnalysis;
+﻿using System.Diagnostics;
+using Microsoft.CodeAnalysis;
 
 namespace TypeShape.Roslyn;
 
 public partial class TypeDataModelGenerator
 {
-    private static bool TryMapEnum(ITypeSymbol type, out TypeDataModel? model, out TypeDataModelGenerationStatus status)
+    private bool TryMapEnum(ITypeSymbol type, ref TypeDataModelGenerationContext ctx, out TypeDataModel? model, out TypeDataModelGenerationStatus status)
     {
         if (type.TypeKind is not TypeKind.Enum)
         {
@@ -13,13 +14,16 @@ public partial class TypeDataModelGenerator
             return false;
         }
 
+        INamedTypeSymbol underlyingType = ((INamedTypeSymbol)type).EnumUnderlyingType!;
+        status = IncludeNestedType(underlyingType, ref ctx);
+        Debug.Assert(status is TypeDataModelGenerationStatus.Success);
+
         model = new EnumDataModel
         {
             Type = type,
-            UnderlyingType = ((INamedTypeSymbol)type).EnumUnderlyingType!,
+            UnderlyingType = underlyingType,
         };
-
-        status = TypeDataModelGenerationStatus.Success;
+        
         return true;
     }
 }

--- a/src/TypeShape.Roslyn/ModelGenerator/TypeDataModelGenerator.cs
+++ b/src/TypeShape.Roslyn/ModelGenerator/TypeDataModelGenerator.cs
@@ -160,7 +160,7 @@ public partial class TypeDataModelGenerator
     /// </remarks>
     protected virtual TypeDataModelGenerationStatus MapType(ITypeSymbol type, ref TypeDataModelGenerationContext ctx, out TypeDataModel? model)
     {
-        if (TryMapEnum(type, out model, out TypeDataModelGenerationStatus status))
+        if (TryMapEnum(type, ref ctx, out model, out TypeDataModelGenerationStatus status))
         {
             return status;
         }

--- a/tests/TypeShape.SourceGenerator.UnitTests/CompilationTests.cs
+++ b/tests/TypeShape.SourceGenerator.UnitTests/CompilationTests.cs
@@ -208,4 +208,21 @@ public static class CompilationTests
         TypeShapeSourceGeneratorResult result = CompilationHelpers.RunTypeShapeSourceGenerator(compilation);
         Assert.Empty(result.Diagnostics);
     }
+    
+    [Fact]
+    public static void EnumGeneration_NoErrors()
+    {
+        // Regression test for https://github.com/eiriktsarpalis/typeshape-csharp/issues/29
+        Compilation compilation = CompilationHelpers.CreateCompilation("""
+           using TypeShape;
+           
+           enum MyEnum { A, B, C }
+
+           [GenerateShape<MyEnum>]
+           partial class Witness { }
+           """);
+
+        TypeShapeSourceGeneratorResult result = CompilationHelpers.RunTypeShapeSourceGenerator(compilation);
+        Assert.Empty(result.Diagnostics);
+    }
 }

